### PR TITLE
feat(config): Add config struct

### PR DESCRIPTION
**Pull request by hendriknielaender**
Introduce a new configuration option in `zbench` that allows users to specify
the number of iterations for benchmarks. This update modifies the `run` function
to prioritize user-defined iteration count from `Config.iterations` over the
automatic calculation based on minimum duration. This enhancement provides
greater flexibility in benchmarking by accommodating specific iteration requirements.

### CODE REVIEW

**Comment by hendriknielaender** [comment]
@Bryysen @FObersteiner Please have a look.

File: zbench.zig [hunk 1]
```diff
@@ -8,6 +8,13 @@ const log = std.log.scoped(.zbench);
 const c = @import("./util/color.zig");
 const format = @import("./util/format.zig");
 
+pub const Config = struct {
+    iterations: u16 = 0,
+    max_iterations: u16 = 16384,
+    budget: u64 = 2e9, // 2 seconds
```
Comment (Bryysen) [inline zbench.zig:14]: Sorry if this is overly nitpicky, but why the name `budget` and not `max_time` or something similar to indicate we are talking about time? Maybe something like `time_budget`?

**Review by Bryysen — COMMENTED** [review]
Overall I agree with this. I was going for something similar with my proposed [RunArgs](https://github.com/Bryysen/zBench/blob/f35b8fe159759ce4ad65e450c5c10c7b514fea74/zbench.zig#L20C5-L30C7) but this works just as well.

File: zbench.zig (hunk 1 above)
Comment (FObersteiner) [reply zbench.zig:14]: > Sorry if this is overly nitpicky, but why the name `budget` and not `max_time` or something similar to indicate we are talking about time? Maybe something like `time_budget`?

I was about to write a similar comment ;-) I think it's important to first clarify what user-defined boundaries should there be and which one takes precedence if their implications overlap. Then the proper naming might become more obvious.

Comment (hendriknielaender) [reply zbench.zig:14]: true 👍 will apply that.

File: zbench.zig [hunk 2]
```diff
@@ -8,6 +8,13 @@ const log = std.log.scoped(.zbench);
 const c = @import("./util/color.zig");
 const format = @import("./util/format.zig");
 
+pub const Config = struct {
+    iterations: u16 = 0,
```
[the remark this answers is not in the document]
Comment (hendriknielaender) [reply zbench.zig:12]: yes

**Pull request merged**